### ISSUE TITLE
Count Example : Error launching source instance: InvalidParameterCombination: Virtualization type 'hvm' is required for instances of type 't2.small'.

### ISSUE THREAD

**Issue by snowch** [terraform-providers/terraform-provider-aws#15139]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* Count Example (commit 4e7e9fb44c5d9c3ca2520d9eae9f141c116ba2b8)

###  Steps to Reproduce

```bash
git clone https://github.com/terraform-providers/terraform-provider-aws.git
cd terraform-provider-aws/examples/count
terraform init
terraform apply
```

### Expected Behavior

Example works

### Actual Behavior

```
Error: Error launching source instance: InvalidParameterCombination: Virtualization type 'hvm' is required for instances of type 't2.small'. Ensure that you are using an AMI with virtualization type 'hvm'. For more information, see http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/virtualization_types.html
	status code: 400, request id: d73d2782-ba24-48c4-822d-813bf34a838a
...
```

**Comment by snowch** [comment]
For now, I have fixed it in my environment by looking up the AMI:

```diff
diff --git a/examples/count/main.tf b/examples/count/main.tf
index 95ed746c7..3eb8ce5da 100644
--- a/examples/count/main.tf
+++ b/examples/count/main.tf
@@ -23,9 +23,25 @@ resource "aws_elb" "web" {
   instances = aws_instance.web.*.id
 }

+data "aws_ami" "ubuntu" {
+  most_recent = true
+
+  filter {
+    name   = "name"
+    values = ["ubuntu/images/hvm-ssd/ubuntu-focal-20.04-amd64-server-*"]
+  }
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+
+  owners = ["099720109477"] # Canonical
+}
+
 resource "aws_instance" "web" {
   instance_type = "t2.small"
-  ami           = var.aws_amis[var.aws_region]
+  ami           = data.aws_ami.ubuntu.id

   # This will create 4 instances
   count = 4
diff --git a/examples/count/variables.tf b/examples/count/variables.tf
index 949160507..3cb51b2e1 100644
--- a/examples/count/variables.tf
+++ b/examples/count/variables.tf
@@ -2,14 +2,3 @@ variable "aws_region" {
   description = "The AWS region to create things in."
   default     = "us-west-2"
 }
-
-# Ubuntu Precise 12.04 LTS (x64)
-variable "aws_amis" {
-  default = {
-    "eu-west-1" = "ami-b1cf19c6"
-    "us-east-1" = "ami-de7ab6b6"
-    "us-west-1" = "ami-3f75767a"
-    "us-west-2" = "ami-21f78e11"
-  }
-}
-
```

**Comment by anGie44** [comment]
Hi @snowch, thank you for creating this issue! Looks like we'll need to update our hardcoded `aws_amis` variable or apply data source filtering as you've described. If you'd like to submit a PR with your fix, we'd be happy to review it as well :)

**Comment by anGie44** [comment]
The fix has been merged and will release with `v3.7.0` of the Terraform AWS Provider, likely out this Thursday.
